NetworkLayerTests/CoreAdvertsServiceTest.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  CoreAdvertServiceTest.swift
+//  NetworkLayerTests
+//
+//  Created by Clément Garcia on 17/10/2022.
+//
+
+import Foundation
+
+import XCTest
+@testable import NetworkLayer
+
+class CoreAdvertsServiceTest: XCTestCase {
+    
+    // MARK: - Vars
+    //Fake session confirguration set in order to preform unit tests
+    private let sessionConfiguration: URLSessionConfiguration = {
+        let sessionConfiguration=URLSessionConfiguration.ephemeral
+        sessionConfiguration.protocolClasses=[URLProtocolFake.self]
+        return sessionConfiguration
+    }()
+    
+    // MARK: - Tests
+    
+    func testGivenCoreServiceIsNeeded_WhenCoreServiceIsRequested_ThenCoreIsProvided() {
+        let sut = CoreAdvertsService()
+        XCTAssertNotNil(sut)
+    }
+    
+    func testGivenNeedAdvertsList_WhenNoIssueOccured_AdvertsListProvided() async {
+        URLProtocolFake.fakeURLs = [ApiEndpoint.getEndpointAdvert(): (FakeResponseData.correctAdvertsListData, FakeResponseData.validResponseCode, nil)]
+        
+        let fakeSession = URLSession(configuration: sessionConfiguration)
+        let sut = CoreAdvertsService(session: fakeSession)
+        
+        do {
+            let sutAdvertsList = try await sut.getAdvertsList()
+            XCTAssertTrue(!sutAdvertsList.isEmpty)
+            print(sutAdvertsList.count)
+        } catch {
+            XCTFail("\(error)")
+        }
```

Comment:
```suggestion
        do {
            let sutAdvertsList = try await sut.getAdvertsList()
            XCTAssertTrue(!sutAdvertsList.isEmpty)
        } catch {
            XCTFail("\(error)")
        }
```
Ok ici vérifier que la liste n'est pas vide ne suffit pas, il faut vérifier si quelques valeurs de la liste sont correctes.